lib/client/src/client/client.ts

```diff
@@ -112,6 +123,54 @@ export interface AuthInfo {
  * @template R The type to use for ranges (such as `vscode.Range` for VS Code extensions).
  */
 export interface Client<R extends Range> {
+    /**
+     * Get the capabilities returned by the configured providers.
+     *
+     * It does not continue to listen for changes, as {@link Client.capabilitiesChanges} does. Using
+     * {@link Client.capabilities} is simpler and does not require use of observables (with a library like
+     * RxJS), but it means that the client needs to manually poll for updated item kinds if freshness is
+     * important.
+     */
+    capabilities(
+        params: CapabilitiesParams,
+        providerUri?: string
+    ): Promise<EachWithProviderUri<CapabilitiesResult[]>>
+
+    /**
+     * Observe OpenCtx capabilities of the configured providers.
+     *
+     * The returned observable streams capabilities as they are received from the providers and continues
+     * passing along any updates until unsubscribed.
+     */
+    capabilitiesChanges(
+        params: CapabilitiesParams,
+        providerUri?: string
+    ): Observable<EachWithProviderUri<CapabilitiesResult[]>>
+
+    /**
+     * Get the candidate items returned by the configured providers.
+     *
+     * It does not continue to listen for changes, as {@link Client.candidateItemsChanges} does. Using
+     * {@link Client.candidateItems} is simpler and does not require use of observables (with a library like
+     * RxJS), but it means that the client needs to manually poll for updated items if freshness is
+     * important.
+     */
+    candidateItems(
+        params: CandidateItemsParams,
+        providerUri?: string
+    ): Promise<EachWithProviderUri<CandidateItemsResult>>
+
+    /**
+     * Observe OpenCtx candidate items from the configured providers.
+     *
+     * The returned observable streams candidate items as they are received from the providers and continues
+     * passing along any updates until unsubscribed.
+     */
+    candidateItemsChanges(
```

Comment:
Motivation:

1. I think we want a single API call for non-interactive applications that gets fully resolved items. Annoying if they have to call 2 APIs and make 2 roundtrips to do that. Can mean the server has discarded its internal state that would have made it just a tiny bit slower to get the fully resolved data, and instead the server needs to do 2x the number of API calls. We want to avoid that in our API design.
2. Many providers are just as fast returning candidate/preview items as they are the full items. Or the client will not need some of the omitted fields. It'd be nice for it to be a pure optimization, not a fundamental API concept, for the provider to return candidate/preview items.

My proposal:

1. For now, we just have `items` and no `candidateItems`. Yes, this means we will get back some data that might've been slow to fetch and is unnecessary to show in the @-mention menu. But it's simpler and probably fine for now.
2. When needed, as an optimization, then we can introduce a candidate items concept. Right now without seeing the use cases fully I am leaning toward channeling it all through the `items` method and letting you call `items` again with a set of possibly dehydrated/partial candidate items and asking it to fully hydrate them. In that way, the candidate items param to `items()` is like a query, so we can reuse an API concept there since we already have a query.